addons/linecompass/functions/fnc_cacheLoop.sqf

```diff
@@ -0,0 +1,105 @@
+#include "..\script_component.hpp"
+
+[{ call FUNC(cacheLoop); }, [], 0.5] call CBA_fnc_waitAndExecute;
+
+private _player = call CBA_fnc_currentUnit;
+
+private _shouldShowCompass = GVAR(Enabled) && { [_player] call EFUNC(main,canHudBeShown) } && { GVAR(CompassAvailableShown) && [_player] call EFUNC(main,getCompass) isNotEqualTo "" };
+
+if (!_shouldShowCompass) exitWith {
+    if (GVAR(CompassShown)) then {
+        call FUNC(HideCompass);
+    };
+};
+
+if (_shouldShowCompass) then {
+    if (!GVAR(CompassShown)) then {
+        call FUNC(ShowCompass);
+    };
+};
+
+if (customWaypointPosition isNotEqualTo GVAR(customWaypointPosition)) then {
+    if (customWaypointPosition isEqualTo []) then {
+        "CUSTOM_WAYPOINT_POSITION" call FUNC(removeLineMarker);
+    } else {
+        ["CUSTOM_WAYPOINT_POSITION", customWaypointPosition, GVAR(CustomWaypointColor)] call FUNC(addLineMarker);
+    };
+    GVAR(customWaypointPosition) = customWaypointPosition;
+};
+
+
+private _unitsToRender = units group _player - [_player];
+if !(isNil "diwako_dui_special_track" && { diwako_dui_special_track isEqualType [] }) then {
+    _unitsToRender append diwako_dui_special_track;
+};
+
+_unitsToRender = _unitsToRender - [objNull];
+
+GVAR(RenderData) = _unitsToRender apply {
+
+    ([_x, _player] call FUNC(getUnitIcon)) params [["_icon", "a3\ui_f\data\map\Markers\Military\dot_ca.paa", [""]], ["_size", 2, [0]]];
+    _size = [PX(_size), PY(_size)];
+
+    private _color = +(_x getVariable [QEGVAR(main,compass_color), [1, 1, 1]]);
+    if (_color isEqualTo [1, 1, 1]) then {
+        _color = +GVAR(DefaultIconColor);
+    };
+
+    if (GVAR(enableOcclusion)) then {
+        private _vis = [vehicle _x, "VIEW"] checkVisibility [eyePos _player,  AGLToASL (_x modelToWorldVisual (_x selectionPosition "Spine2"))];
+
+        if (_vis isNotEqualTo 0) then {
+            _x setVariable [QGVAR(lastSeen), time + 0.5];
+        };
+    };
```

Comment:
Imo better to have this in the actual PFH. Else everyone gets checked at the same time pops up at the same time as well.
This also helps to spread out the check on different frames to prevent stuttering every 0.5 seconds